docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+        ONTO m1vktopcva7l6spiinh5e5nnc4dtje4ygw2fhismbmczbyaqbws7jq
+    {
+    ALTER TYPE default::User {
+        CREATE PROPERTY friend_names: array<std::str>;
+    };
+    update User set { friend_names := array_agg(.friends.name) };
+    };
+
+Now if we do a query we can confirm that the data inside ``.friends.name`` 
+when converted to an array is indeed the same as the data inside the 
+``friend_names`` property:
+
+.. code-block:: edgeql-repl
+
+    db> select User { f:= array_agg(.friends.name), friend_names };
+    {
+    default::User {
+      f: ['User 2', 'User 3'], 
+      friend_names: ['User 2', 'User 3']
+      },
+    default::User {
+      f: ['User 1', 'User 3'], 
+      friend_names: ['User 1', 'User 3']
+      },
+    default::User {
+      f: ['User 1', 'User 2'], 
+      friend_names: ['User 1', 'User 2']
+      },
+    }
+
+Or if we can't eyeball the data ourselves, we can use the ``all()`` 
+function to ensure that this is the case:
+
+.. code-block:: edgeql-repl
+
+    db> select all(array_agg(User.friends.name) = User.friend_names);
+    {true}
+
+Looks good! And now we can simply remove ``multi friends: User;`` 
+from our schema and do a final migration.
+
+Migration internals
+===================
+
+We've now reached the most optional part of the migrations tutorial, 
+but an interesting one for those curious about what goes on behind 
+the scenes during a migration.
+
+Migrations in EdgeDB before the advent of the EdgeDB Project flow 
+were still automated, but required more manual work if you didn't 
+want to accept all of the suggestions provided by the server. This 
+process is in fact still used to migrate even today; the CLI just 
+facilitates it by making it easy to respond to the generated suggestions.
+
+They took place `inside a transaction <transaction_>`_ handled by 
+the user that essentially went like this: 
+
+.. code-block::
+
+    db> start migration to { <your schema goes here> };
+
+This starts the migration, after which the quickest process was to 
+type ``populate migration`` to accept the statements suggested by 
+the server, and then ``commit migration`` to finish the process.
+
+Now, there is another option besides simply typing ``populate migration`` 
+that allows you to look at and handle the suggestions every step of 
+the way (in the same way the CLI does today), and this is what we 
+are going to have some fun with. You can see `the original RFC <rfc_>`_ 
+for this if you are curious.
+
+It is *very* finicky compared to the CLI, resulting in a failed transaction 
+if any step along the way is different from the expected behavior, 
+but is an entertaining challenge to try to get right if you want to 
+truly understand how migrations work in EdgeDB.
+
+This process requires looking at the server's proposed solutions every 
+step of the way, and these steps are best seen in JSON format. Let's 
+first set the CLI to make the format nicely readable with this command:
```

Comment:
We should tell readers to start a new project before we get to this point if they want to follow along. (I assume that's the intention here anyway is for them to have a fresh project.)

Reply:
I don't want to give the impression that using the old method to migrate a schema requires a complete refresh of anything (the lesson to take home is that this is simply what happens unseen to the user every time the CLI does a migration for you), but you've reminded me that the reader won't know how to destroy and restart an instance when necessary. So added that here: https://github.com/edgedb/edgedb/pull/6656/commits/986d069ef6db99e83f49cee35e1c90f1ac6234b2